### PR TITLE
feat(analysis): Add Jenks tier generator with validation

### LINKED ISSUE #18: [FEATURE] [Knob Importance] Data-driven tier generation (Jenks + silhouette)
## [Knob Importance] Data-driven tier generation (Jenks + silhouette)

**Parent issue:** Knob Importance Analysis (fANOVA + SHAP)

### Problem

Our current tier boundaries (5 minimal / 13 core / ~30 standard / all extensive) are fixed by expert judgment. We need an empirical method to determine both the optimal number of tiers and their boundaries from observed importance scores.

### Goal

Convert continuous fANOVA importance scores into discrete tiers using Jenks Natural Breaks clustering, with silhouette score validation to determine the optimal tier count.

#### New file: `src/analysis/tier_generator.py`

**Silhouette validation:**

- For `k` in `{2, 3, 4, 5, 6}`, apply `jenkspy.jenks_breaks()` on the marginal importance scores to get break points. Assign each knob to a tier based on which interval it falls in. Compute `sklearn.metrics.silhouette_score()`.
- Select `k_optimal` that maximizes silhouette.
- Edge case: all importances equal → silhouette is undefined → fall back to `k=3` with a warning.

**Tier naming:**

- `k=3` → `{"minimal", "standard", "extensive"}`
- `k=4` → `{"minimal", "core", "standard", "extensive"}`
- Other `k` → `{"tier_1", ..., "tier_k"}` with a logged warning.

**Per-workload analysis:**

- Accept a `workload_label` parameter (e.g., `"oltp"`, `"olap"`) that tags the output.
- Cross-workload comparison is done via a **separate `--compare` mode** in the CLI (Sub-Issue 7): the user runs the analysis once per workload, then runs `--compare oltp_results.json olap_results.json` to produce a comparison report highlighting knobs that shift tiers across workloads.

**Expert comparison:**

- Compare data-driven tier assignments against expert tiers from `KNOB_TUNING_METADATA[knob].impact_tier`.
- Produce an agreement/disagreement report:
  - `agreements: List[str]` — knobs where expert and data agree.
  - `promotions: List[Tuple[str, str, str]]` — `(knob, expert_tier, data_tier)` where data says the knob is MORE important than the expert thought.
  - `demotions: List[Tuple[str, str, str]]` — where data says LESS important.

**Return type:** `TierResult` dataclass:

- `optimal_k: int`
- `silhouette_scores: Dict[int, float]`
- `tier_assignments: Dict[str, str]` (knob → tier name)
- `jenks_breaks: List[float]`
- `agreement_report: AgreementReport`
- `workload_label: str`

### Files

| Action | File                                         |
| ------ | -------------------------------------------- |
| Create | `src/analysis/tier_generator.py`             |
| Create | `tests/unit/analysis/test_tier_generator.py` |

### Tests

- Silhouette computed for all `k` values; correct optimal `k` returned for known-clustered data (e.g., 3 clearly separated groups → `k=3`).
- Jenks correctly separates clearly bimodal importance scores into 2 tiers.
- Tier names map correctly for `k=3` and `k=4`.
- Agreement report correctly identifies a promotion (expert="extensive", data="minimal").
- Agreement report correctly identifies a demotion.
- All-equal importances → fallback to `k=3` with a warning.
- Single knob → single tier, no crash.

### Acceptance Criteria

- [ ] Optimal `k` is determined via silhouette validation.
- [ ] Jenks Natural Breaks produces correct tier boundaries.
- [ ] Expert comparison report is generated.
- [ ] All edge cases handled.
- [ ] All tests pass.

### PR DESCRIPTION
**Pull request by mohamed20o03**
Use Jenks Natural Breaks with silhouette selection to derive importance tiers. Include expert agreement report, workload tagging, compare mode, and tests. Add jenkspy dependency.

Closes #18

### CODE REVIEW

File: pg_instances/sysbench/pg_snapshots/pbt-snapshot-sysbench_t10_s100000-9f5edcf191bc.json
Comment (alshawai) [inline pg_instances/sysbench/pg_snapshots/pbt-snapshot-sysbench_t10_s100000-9f5edcf191bc.json:1]: 🔴 The snapshot/instance manifest, _or data_, should **NOT** appear on Git. This shall be removed - instances and snapshot folder has already been migrated to `.instances/` and `.snapshots/`, and correctly included in `.gitignore`.

File: .gitignore
Comment (alshawai) [inline .gitignore:1]: 🔴 `.gitignore` **SHOULD** be tracked. We shouldn't hard code file exclusions in `.gitignore`. And, as mentioned before, instances and snapshots location have been migrated to another folder, and `.gitignore` correctly addresses that. _**(Revert all changes to `.gitignore`.)**_

File: src/analysis/importance.py [hunk 1]
```diff
@@ -51,33 +49,6 @@ class InsufficientDataError(Exception):
 class ImportanceResult:
     """
     Container for fANOVA importance variance decomposition results.
-
```
Comment (alshawai) [inline src/analysis/importance.py:54]: 🟡 Docstrings are important for reviewers and contributors so they can easily understand what a module, class, or a function does... **Revert this change to bring back docstrings.**

File: src/analysis/importance.py [hunk 2]
```diff
@@ -89,9 +60,6 @@ class ImportanceResult:
     shap_importances: dict[str, float]
     shap_values: np.ndarray
     fanova_shap_correlation: float
-    scoring_policy: str = "fixed_v1"
```
Comment (alshawai) [inline src/analysis/importance.py:92]: **🔴 UNEXPECTED BEHAVIOR:** scoring policy metadata have been incorrectly removed from `ImportanceResult`. **Revert this change.**

**Review by alshawai — COMMENTED** [review]
## 🔴 Action Required

Many unexpected code edits have been made - We assume it was a mistake by the contributor. There's also **no results provided about `tier_generator.py` script output - It's advised to add result output in your PR or comments for further validation.**

> _More details in comments below._

**Comment by mohamed20o03** [comment]
[tier_result.json](https://github.com/user-attachments/files/27544794/tier_result.json)
I have added the result output in the comments for your review and validation.
